docs/v1/P4Runtime-Spec.mdk

```diff
@@ -5472,6 +5472,78 @@ the purpose of adding features for the P4Runtime API.
 | `@p4runtime_translation` | See sections [#sec-user-defined-types], [#sec-translation-of-port-numbers] |
 +--------------------------+---------------------------------------+
 ~
+### Custom Annotations in P4Info
+P4 allows flexible annotations. Users may add their own annotations as long as they obey the language syntax.
+To facilitate consumption of user-defined annotations, P4Info has an `anno_map` for any element which can accept
+such annotations, e.g. the `PkgInfo`, `Preamble`, `MatchField`, `ActionRef`, `Action` and
+`ControllerPacketMetadata` messages. The `anno_map` contains a suitable parsed representation for the different
+types of annotation bodies, namely: empty, free-form string, and a key-value-map.
+
+Given the following P4 source fragment:
+
+~ Begin P4Example
+    @my_anno1()
+    @my_anno2("Some random content: for i in range(10)")
+    @my_anno3(a=b,c=d)
+    @my_anno3(e=f)
+    table test1 {
```

Comment:
Can we add one more example? I believe the following one is legit (but useless) P4:
```
@my_anno4(1+1, "Hello", 1 << 3)
```
I believe that with this proposal we would use a free-form string for this.

Reply:
Agreed, will do.